packages/app/src/ui/components/ConfirmRequestModal/components/ConnectionModal/ConnectionModal.tsx

```diff
@@ -0,0 +1,148 @@
+import { IPendingRequest } from "@cryptkeeperzk/types";
+import Box from "@mui/material/Box";
+import Tab from "@mui/material/Tab";
+import Tabs from "@mui/material/Tabs";
+import Typography from "@mui/material/Typography";
+import { NavLink } from "react-router-dom";
+
+import { ButtonType, Button } from "@src/ui/components/Button";
+import { FullModal, FullModalFooter, FullModalHeader } from "@src/ui/components/FullModal";
+import { IdentityList } from "@src/ui/components/IdentityList";
+import { SiteFavicon } from "@src/ui/components/SiteFavicon/SiteFavicon";
+
+import { CreateIdentityModal } from "../createIdentityModal";
+
+import "./connectIdentity.scss";
+import { EConnectIdentityTabs, useConnectIdentity } from "./useConnectionModal";
+
+export interface IConnectIdentityModalProps {
+  len: number;
+  loading: boolean;
+  error: string;
+  pendingRequest: IPendingRequest<{ urlOrigin: string }>;
+  accept: () => void;
+  reject: () => void;
+}
+
+export const ConnectIdentityModal = ({
+  len,
+  pendingRequest,
+  error,
+  loading,
+  accept,
+  reject,
+}: IConnectIdentityModalProps): JSX.Element => {
+  const {
+    linkedIdentities,
+    unlinkedIdentities,
+    urlOrigin,
+    faviconUrl,
+    selectedTab,
+    selectedIdentityCommitment,
+    onTabChange,
+    onSelectIdentity,
+    openCreateIdentityModal,
+    onCreateIdentityModalShow,
+    onReject,
+    onAccept,
+  } = useConnectIdentity({
+    pendingRequest,
+    accept,
+    reject,
+  });
+
+  return (
+    <div>
+      {openCreateIdentityModal ? (
+        <CreateIdentityModal
+          connectIdentityCallback={onCreateIdentityModalShow}
+          reject={reject}
+          urlOrigin={urlOrigin}
+        />
+      ) : (
+        <FullModal data-testid="connect-identity-page" onClose={onReject}>
```

Comment:
This one needs to be removed from ConfirmRequestModal right after this PR. Right now we have two copy-paste components which do the same things.